src/main/java/org/elasticsearch/common/util/concurrent/BaseFuture.java

```diff
@@ -89,6 +92,7 @@
     @Override
     public V get(long timeout, TimeUnit unit) throws InterruptedException,
             TimeoutException, ExecutionException {
+        assert Transports.isTransportThread(Thread.currentThread()) == false : Thread.currentThread().getName();
```

Comment:
may add something like "thread ["+ ..+"] executed a blocking operation but is a network thread" . I think it will make it easier to debug.